standard/patterns.md

```diff
@@ -0,0 +1,176 @@
+- the pattern `P` *matches* the value *e* if `object.Equals(e, v)` returns `true`.
+
+> *Example*:
+>
+> ```csharp
+> public static decimal GetGroupTicketPrice(int visitorCount)
+> {
+>     switch (visitorCount) {
+>         case 1: return 12.0m;
+>         case 2: return 20.0m;
+>         case 3: return 27.0m;
+>         case 4: return 32.0m;
+>         case 0: return 0.0m;
+>         default: throw new ArgumentException(…);
+>     }
+> }
+> ```
+>
+> *end example*
+
+### §var-pattern-new-clause Var pattern
+
+A *var_pattern* matches every value.  That is, a pattern-matching operation with a *var_pattern* always succeeds.
+
+A *var_pattern* is *applicable to* every type.
+
+```ANTLR
+var_pattern
+    : 'var' designation
+    ;
+designation
+    : simple_designation
+    ;
+```
+
+Given a *pattern input value* *e*, if *designation* is the *identifier* `_`, it denotes a discard (§discards-new-clause), and the value of *e* is not bound to anything. (Although a declared variable with that name may be in scope at that point, that named variable is not seen in this context.) If *designation* is any other identifier, at runtime the value of *e* is bound to a newly introduced local variable ([§9.2.8](variables.md#928-local-variables)) of that name whose type is the static type of *e*, and the pattern input value is assigned to that local variable.
+
+It is an error if the name `var` would bind to a type where a *var_pattern* is used.
+
+## Pattern Subsumption
+
+In a switch statement, it is an error if a case's pattern is *subsumed* by the preceding set of unguarded cases (XREF).
+Informally, this means that any input value would have been matched by one of the previous cases.
+Here we define when a set of patterns *subsumes* a given pattern.
+
+We say a pattern `P` *would match* a constant `K` if the specification for that pattern's runtime behavior is that `P` matches `K`.
+
+A set of patterns `Q` *subsumes* a pattern `P` if any of the following conditions hold:
+
+- `P` is a constant pattern and any of the patterns in the set `Q` would match `P`'s *converted value*
+- `P` is a var pattern and the set of patterns `Q` is *exhaustive* for the type of the pattern input value, and either the pattern input value is not of a nullable type or some pattern in `Q` would match `null`.
+- `P` is a declaration pattern with type `T` and the set of patterns `Q` is *exhaustive* for the type `T` (XREF).
+
+## Pattern Exhaustiveness
+
+Informally, we say that a set of patterns is exhaustive for a type if some pattern in the set is applicable to every possible value of that type other than null.
+Here we define when a set of patterns is *exhaustive* for a type.
+
+A set of patterns `Q` is *exhaustive* for a type `T` if any of the following conditions hold:
+1. `T` is an integral or enum type, or a nullable version of one of those, and for every possible value of `T`'s underlying type, some pattern in `Q` would match that value; or
```

Comment:
"underlying" is meant to apply to nullable types, not enum types.

Reply:
change "underlying" to "nullable underlying".

Reply:
added in #934